routing-java-ortools/src/main/java/com/nextmv/example/Main.java

```diff
@@ -0,0 +1,110 @@
+
+    Loader.loadNativeLibraries();
+
+    // Create Routing Index Manager
+    RoutingIndexManager manager = new RoutingIndexManager(input.distanceMatrix.length, input.vehicleNumber,
+        input.depot);
+
+    // Create Routing Model.
+    RoutingModel routing = new RoutingModel(manager);
+
+    // Create and register a transit callback.
+    final int transitCallbackIndex = routing.registerTransitCallback((long fromIndex, long toIndex) -> {
+      // Convert from routing variable Index to user NodeIndex.
+      int fromNode = manager.indexToNode(fromIndex);
+      int toNode = manager.indexToNode(toIndex);
+      return input.distanceMatrix[fromNode][toNode];
+    });
+
+    // Define cost of each arc.
+    routing.setArcCostEvaluatorOfAllVehicles(transitCallbackIndex);
+
+    // set the duration of the search
+    Duration duration = Duration.newBuilder().setSeconds(options.getDuration()).build();
+
+    // Setting first solution heuristic.
+    RoutingSearchParameters searchParameters = main.defaultRoutingSearchParameters()
+        .toBuilder()
+        .setFirstSolutionStrategy(FirstSolutionStrategy.Value.PATH_CHEAPEST_ARC)
+        .setTimeLimit(duration)
+        .build();
+
+    // Solve the problem.
+    // record solve start time
+    long solveStartTime = System.currentTimeMillis();
+    Assignment solution = routing.solveWithParameters(searchParameters);
+
+    Output output = getOutput(startTime, input, routing, manager, solution, solveStartTime);
+
+    // Write output. Exit on error.
+    Output.write(options.getOutputPath(), output);
+  }
+
+  static Output getOutput(
+      long startTime,
+      Input input,
+      RoutingModel routing,
+      RoutingIndexManager manager,
+      Assignment solution, long solveStartTime) {
+    long maxRouteDistance = 0;
+    List<Vehicle> vehicles = new ArrayList<Vehicle>();
+    for (int i = 0; i < input.vehicleNumber; ++i) {
+      List<Integer> stops = new ArrayList<Integer>();
+      long index = routing.start(i);
+      long routeDistance = 0;
+      while (!routing.isEnd(index)) {
+        stops.add(manager.indexToNode(index));
+        long previousIndex = index;
+        index = solution.value(routing.nextVar(index));
+        routeDistance += routing.getArcCostForVehicle(previousIndex, index, i);
+      }
+      stops.add(manager.indexToNode(index));
+      Vehicle vehicle = new Vehicle(i, routeDistance, stops);
+      vehicles.add(vehicle);
+      maxRouteDistance = Math.max(routeDistance, maxRouteDistance);
+    }
+
+    // compute solve duration
+    long endTime = System.currentTimeMillis();
+    double duration = endTime - solveStartTime;
+    // convert duration to seconds
+    duration = duration / 1000.0;
+
+    // compute total duration
+    endTime = System.currentTimeMillis();
+    double runDuration = endTime - startTime;
+    // convert duration to seconds
+    runDuration = runDuration / 1000.0;
+
+    // create output
```

Comment:
```suggestion
    // Compute solve duration.
    long endTime = System.currentTimeMillis();
    double duration = endTime - solveStartTime;
    // Convert duration to seconds.
    duration = duration / 1000.0;

    // Compute total duration.
    endTime = System.currentTimeMillis();
    double runDuration = endTime - startTime;
    // Convert duration to seconds.
    runDuration = runDuration / 1000.0;

    // Create output.
```

What have I started? :sweat_smile: